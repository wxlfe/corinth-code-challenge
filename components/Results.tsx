import * as React from 'react'
import { Collapse } from '@nextui-org/react'
import Result from './Result'
import { People, Person } from '../interfaces'

type Props = {
  people: People
}

const Results = ({ people }: Props) => (
  <Collapse.Group>
    { people.map((person: Person) => {
      return <Result data={ person } />
    } )}
  </Collapse.Group>
)

export default Results
