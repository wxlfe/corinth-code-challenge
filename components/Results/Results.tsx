import * as React from 'react';
import { Collapse } from '@nextui-org/react';
import Result from '../Result';
import { Person } from '../../interfaces';

type Props = {
  people: Person[];
};

const Results = ({ people }: Props) => (
  <Collapse.Group>
    {people.map((person: Person, index: number) => {
      return <Result data={person} key={index} />;
    })}
  </Collapse.Group>
);

export default Results;
