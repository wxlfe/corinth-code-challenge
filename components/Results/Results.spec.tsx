import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Results from './Results';

describe('Results Component', () => {
  test('should create', () => {
    expect(Results).toBeTruthy();
  });

  test('should render', () => {
    const { asFragment } = render(
      <Results
        people={[
          {
            name: 'string',
            height: 'string',
            mass: 'string',
            hair_color: 'string',
            skin_color: 'string',
            eye_color: 'string',
            birth_year: 'string',
            gender: 'male',
            homeworld: 'string',
            films: [],
            species: [],
            vehicles: [],
            starships: [],
            created: 'string',
            edited: 'string',
            url: 'string',
          },
        ]}
      ></Results>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
