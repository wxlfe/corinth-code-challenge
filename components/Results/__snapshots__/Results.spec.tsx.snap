// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Component should render 1`] = `
<DocumentFragment>
  <div
    class="nextui-c-dWWWSv nextui-c-dWWWSv-eaqZHc-borderWeight-light"
  >
    <div
      class="nextui-c-hPzDAx nextui-c-huiNHE nextui-c-hPzDAx-eLdNZT-borderWeight-light nextui-c-hPzDAx-jBQhgJ-divider-true nextui-c-hPzDAx-kMkCGX-animated-true nextui-c-hPzDAx-TDdeu-visible-false nextui-collapse nextui-collapse--closed"
      data-state="closed"
      tabindex="-1"
    >
      <div
        aria-controls="react-aria9927596532-2"
        aria-disabled="false"
        aria-expanded="false"
        class="nextui-c-lfcDHB nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-collapse-view"
        data-state="closed"
        id="react-aria9927596532-1"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-collapse-title-container"
        >
          <div
            class="nextui-collapse-title-content"
          >
            <h3
              class="nextui-collapse-title"
            >
              string
            </h3>
          </div>
          <div
            class="nextui-collapse-title-content-right"
          >
            <svg
              class="nextui-c-fUXRSc nextui-collapse-icon"
              fill="none"
              focusable="false"
              height="20"
              role="presentation"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.5 19l-7-7 7-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="nextui-c-dJMBOt nextui-c-dJMBOt-ielLSBX-css"
      >
        <div
          class="nextui-expand-content"
        >
          <div
            aria-labelledby="react-aria9927596532-1"
            class="nextui-c-fMAxGR nextui-collapse-content"
            id="react-aria9927596532-2"
            role="region"
            tabindex="-1"
          >
            <div
              style="display: grid; grid-template-columns: repeat(auto-fit, minmax(320px, 1fr));"
            >
              <div
                class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-PJLV nextui-c-BDLTQ-iDQuvU-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-PJLV-gulvcB-isFocusVisible-false"
                role="section"
                tabindex="-1"
              >
                <div
                  class="nextui-c-jxECYO"
                >
                  <h3
                    aria-label="About Me"
                    class="nextui-c-PJLV nextui-c-PJLV-ijnlmIz-css"
                  >
                    About Me
                  </h3>
                </div>
                <div
                  class="nextui-c-bZLhCa nextui-c-bZLhCa-qCmpO-color-default nextui-c-bZLhCa-iiMwZzC-css"
                  role="separator"
                />
                <div
                  class="nextui-c-eFfoBo"
                >
                  <div
                    class="nextui-c-hWZRae nextui-c-hWZRae-gikTHb-shadow-true nextui-c-hWZRae-bzEnZV-borderWeight-normal nextui-c-hWZRae-iPJLV-css nextui-table-container"
                  >
                    <table
                      aria-describedby=""
                      aria-label="About Me Table"
                      class="nextui-c-fcXuxO nextui-c-fcXuxO-fyxmJu-color-primary nextui-c-fcXuxO-fPgLZC-lineWeight-light nextui-table"
                      id="react-aria9927596532-3"
                      role="grid"
                      tabindex="0"
                    >
                      <thead
                        class="nextui-c-PJLV nextui-table-row-group"
                        role="rowgroup"
                      >
                        <tr
                          class="nextui-c-fRbcxI nextui-table-header-row"
                          role="row"
                        >
                          <th
                            class="nextui-c-kNgzEX nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-column-header"
                            data-key=".0.0"
                            id="react-aria9927596532-3-.0.0"
                            role="columnheader"
                            tabindex="-1"
                          >
                            Attribute
                          </th>
                          <th
                            class="nextui-c-kNgzEX nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-column-header"
                            data-key=".0.1"
                            id="react-aria9927596532-3-.0.1"
                            role="columnheader"
                            tabindex="-1"
                          >
                            Value
                          </th>
                        </tr>
                        <tr
                          aria-hidden="true"
                          class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-icCJVMZ-css nextui-table-hidden-row"
                        />
                      </thead>
                      <tbody
                        class="nextui-c-PJLV nextui-c-PJLV-iccTTUn-css nextui-table-body"
                        role="rowgroup"
                      >
                        <tr
                          aria-hidden="false"
                          aria-labelledby="react-aria9927596532-3-1-.0.0"
                          class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
                          data-key="1"
                          role="row"
                          tabindex="-1"
                        >
                          <td
                            class="nextui-c-PJLV nextui-c-czOCAM nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
                            data-key="1.0"
                            id="react-aria9927596532-3-1-.0.0"
                            role="rowheader"
                            tabindex="-1"
                          >
                            Height: 
                          </td>
                          <td
                            class="nextui-c-PJLV nextui-c-czOCAM nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
                            data-key="1.1"
                            role="gridcell"
                            tabindex="-1"
                          >
                            string
                          </td>
                        </tr>
                        <tr
                          aria-hidden="false"
                          aria-labelledby="react-aria9927596532-3-2-.0.0"
                          class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
                          data-key="2"
                          role="row"
                          tabindex="-1"
                        >
                          <td
                            class="nextui-c-PJLV nextui-c-czOCAM nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
                            data-key="2.0"
                            id="react-aria9927596532-3-2-.0.0"
                            role="rowheader"
                            tabindex="-1"
                          >
                            Mass: 
                          </td>
                          <td
                            class="nextui-c-PJLV nextui-c-czOCAM nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
                            data-key="2.1"
                            role="gridcell"
                            tabindex="-1"
                          >
                            string
                          </td>
                        </tr>
                        <tr
                          aria-hidden="false"
                          aria-labelledby="react-aria9927596532-3-3-.0.0"
                          class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
                          data-key="3"
                          role="row"
                          tabindex="-1"
                        >
                          <td
                            class="nextui-c-PJLV nextui-c-czOCAM nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
                            data-key="3.0"
                            id="react-aria9927596532-3-3-.0.0"
                            role="rowheader"
                            tabindex="-1"
                          >
                            Hair Color: 
                          </td>
                          <td
                            class="nextui-c-PJLV nextui-c-czOCAM nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
                            data-key="3.1"
                            role="gridcell"
                            tabindex="-1"
                          >
                            string
                          </td>
                        </tr>
                        <tr
                          aria-hidden="false"
                          aria-labelledby="react-aria9927596532-3-4-.0.0"
                          class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
                          data-key="4"
                          role="row"
                          tabindex="-1"
                        >
                          <td
                            class="nextui-c-PJLV nextui-c-czOCAM nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
                            data-key="4.0"
                            id="react-aria9927596532-3-4-.0.0"
                            role="rowheader"
                            tabindex="-1"
                          >
                            Birth Year: 
                          </td>
                          <td
                            class="nextui-c-PJLV nextui-c-czOCAM nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
                            data-key="4.1"
                            role="gridcell"
                            tabindex="-1"
                          >
                            string
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
              <div
                class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-PJLV nextui-c-BDLTQ-iDQuvU-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-PJLV-gulvcB-isFocusVisible-false"
                role="section"
                tabindex="-1"
              >
                <div
                  class="nextui-c-jxECYO"
                >
                  <h3
                    aria-label="Films Appeared In"
                    class="nextui-c-PJLV nextui-c-PJLV-ijnlmIz-css"
                  >
                    Films Appeared In
                  </h3>
                </div>
                <div
                  class="nextui-c-bZLhCa nextui-c-bZLhCa-qCmpO-color-default nextui-c-bZLhCa-iiMwZzC-css"
                  role="separator"
                />
                <div
                  class="nextui-c-eFfoBo"
                >
                  <div
                    style="display: grid; grid-template-columns: repeat(auto-fit, minmax(90px, 1fr));"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
